rfcs/20210510-binary-size-reduction/ISA_extension.md

```diff
@@ -0,0 +1,117 @@
+# Build knobs controlling binary size extension.
+
+## Problem Statement.
+This follows up this
+[RFC](https://github.com/oneapi-src/oneDNN/tree/rfcs/rfcs/20210510-binary-size-reduction)
+and proposes additional build knob to control functionality by ISA. It may
+provide additional save up to 5 MB, which is around ten percent (~10%) from
+current full binary size, based on selected ISA.
+
+## Proposal
+The proposal is to introduce one more build option, `DNNL_ENABLE_PRIMITIVE_ISA`.
+It would take effect if `DNNL_X64` build setting was specified.
+
+Library supported ISA are only partially ordered, that is why the suggestion is
+to group them, so that these groups are in linear order. Here they are:
+* `SSE41`  - supports sse41 ISA only and all compiler-based optimized
+             implementations.
+* `AVX2`   - supports `SSE41` group and avx and avx2 ISA extensions on top.
+* `AVX512` - supports `AVX2` group and most AVX512 ISA extensions: avx512_mic,
+             avx512_mic_4ops, avx512_core, avx512_core_vnni, avx512_core_bf16
+             and avx2_vnni.
+* `ALL`    - supports `AVX512` group and avx512_core_amx ISA extension on top.
```

Comment:
Since the groups are inclusive, the `ALL` group should be an alias for the latest group. In this case `ALL` should be equal to `AVX512`. Why do we add `avx512_core_amx` to `ALL`? It looks like there should be a separate group for `AMX`.

Reply:
Agree with your point. Will fix it.

Reply:
Updated as proposed. Thanks for a suggestion.